File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/BuildMethods.java
package io.github.alexiscomete.lapinousecond.worlds.buildings;

import io.github.alexiscomete.lapinousecond.entity.Player;
import org.javacord.api.entity.message.MessageBuilder;
import org.javacord.api.entity.message.embed.EmbedBuilder;

public interface BuildMethods {

    EmbedBuilder getInfos(Player p);
    MessageBuilder getCompleteInfos(Player p);
    void configBuilding();
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/BuildMethods.kt
package io.github.alexiscomete.lapinousecond.worlds.buildings

import io.github.alexiscomete.lapinousecond.view.ui.longuis.LongCustomUI
import io.github.alexiscomete.lapinousecond.view.ui.playerui.PlayerUI

interface BuildMethods {
    fun getCompleteInfos(ui: PlayerUI): LongCustomUI
    fun configBuilding()
    fun interpret(args: Array<String>)
    val help: String?
    val usage: String?
    fun descriptionShort(): String
    fun title(): String
}